From fef5836e5f1180d124a34fd5a21d58a44137f87e Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@foss.st.com>
Date: Fri, 25 Aug 2023 09:23:48 +0200
Subject: [PATCH 135/140] arm64: dts: st: add NVMEM provider for TAMP on
 STM32MP25

TAMP backup registers allow information sharing between different
software components/execution levels.
They are described individually as nvmem cells.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/linux.git 2c94fcbb803be1c815516c54effb23ebfd5d99ec arm64: dts: st: add NVMEM provider for TAMP on STM32MP25]
Change-Id: I9e1dc6334daecda685244fc3bb231bd9a7d46474
Signed-off-by: Simeon Marijon <simeon.marijon@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 21ef95e262e17..13bd49127a9a0 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2063,6 +2063,25 @@ syscfg: syscon@44230000 {
 			#clock-cells = <0>;
 		};
 
+		tamp: tamp@46010000 {
+			compatible = "st,stm32mp25-tamp", "syscon", "simple-mfd";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			reg = <0x46010000 0x400>;
+			ranges;
+
+			nvram: nvram@46010100 {
+				compatible = "st,stm32mp25-tamp-nvram";
+				#address-cells = <1>;
+				#size-cells = <1>;
+				reg = <0x46010100 0x200>;
+
+				boot_mode: tamp-bkp@180 {
+					reg = <0x180 0x4>;
+				};
+			};
+		};
+
 		pinctrl: pinctrl@44240000 {
 			#address-cells = <1>;
 			#size-cells = <1>;
-- 
2.51.0

